lib/exp.py

```diff
@@ -30,6 +31,37 @@ def __init__(self, namespace='default'):
         self.k8s = client.CustomObjectsApi()
         self.batch = client.BatchV1Api()
 
+    # Helper methods
+
+    def _retry_poll_api(self, api, max_retries_error, max_retries=30,
+                        retry_interval=1, api_kwargs={}):
+        """
+        Helper function that has a polling loop to retry calling the specified
+        API until it's successful (the client does not throw an Api Exception).
+
+        :param api: Client API to call
+        :param max_retries_error: Error message to print if the maximum retries
+                                  has been reached and the API call still fails
+        :param max_retries: Maximum number of times to retry calling the API
+        :param retry_interval: Number of seconds to wait between API retries
+        :param api_kwargs: Dictionary of API arguments
+        :return: Return value of the client API call
+        """
+        retry_count = 0
+        while retry_count < max_retries:
+            try:
+                return api(**api_kwargs)
+            except client.rest.ApiException:
```

Comment:
is there any way to make this more specific or is this the only exception type that gets thrown?  I worry we'll end up in a retry loop in scenarios other than just that the other side hasn't come up yet.

Reply:
I looked at the documentation and it looks like `ApiException` is the only kubernetes exception that they check for.  I looked at their code, and there are other python exceptions used like `TypeError` and `ValueError`, but those aren't things that we'd want to do a retry for.

Reply:
ok, we'll just have to role with this Exception then.